addons/web/static/src/webclient/webclient.js

```diff
@@ -54,9 +54,19 @@ export class WebClient extends Component {
     }
 
     async loadRouterState() {
+        let menuId = Number(router.current.menu_id || 0); // This is for retro-compatibility url
+        const firstAction = router.current.actionStack?.[0]?.action;
+        if (!menuId && firstAction) {
+            menuId = this.menuService
+                .getAll()
+                .find((m) => m.actionID === firstAction || m.actionPath === firstAction)?.appID;
+        }
+        if (menuId) {
+            this.menuService.setCurrentMenu(menuId);
+        }
         let stateLoaded = await this.actionService.loadState();
-        let menuId = Number(router.current.menu_id || 0);
 
+        // Retrocompatibility: when there is only menu_id in url
```

Comment:
would be nice to have some keyword in the retrocompatibility comments, s.t. we can easily grep all of them